src/resample/tests/test_bootstrap.py

```diff
@@ -17,9 +19,54 @@
 f = ecdf(x)
 
 
-def test_jackknife_shape():
-    jack = jackknife(x)
-    assert jack.shape == (n, n - 1)
+def test_jackknife():
+    x = [0, 1, 2, 3]
+    r = jackknife(x, lambda x: x.copy())
+    assert_equal(r, [[1, 2, 3], [0, 2, 3], [0, 1, 3], [0, 1, 2]])
+
+
+def test_jackknife_bias():
+    x = [0, 1, 2, 3]
+    # bias is exactly zero for linear functions
+    r = jackknife_bias(x, np.mean)
+    assert r == 0
+
+    # this "mean" has a bias of exactly O(n^{-1})
+    def bad_mean(x):
+        return (np.sum(x) + 2) / len(x)
+
+    x = [0, 1, 2]
+    r = jackknife_bias(x, bad_mean)
+    mean_loo_f = np.mean([bad_mean([1, 2]), bad_mean([0, 2]), bad_mean([0, 1])])
+    # (5/2 + 4/2 + 3/2) / 3 = 12 / 6 = 2
+    assert mean_loo_f == 2.0
+    # f = 5/3
+    # (n-1) * (mean_loo_f - f)
+    # (3 - 1) * (6/3 - 5/3) = 2/3
+    assert r == pytest.approx(2.0 / 3.0)
```

Comment:
assert_almost_equal would also work. I slightly prefer pytest.approx for testing single numbers, since assert_equal and assert_almost_equal are primarily designed for arrays.

Reply:
pytest.approx is the intended pytest facility for testing floating point results.

Reply:
It is a matter of taste.